File: metrics-api/models/web/mission.go
package web

import (
	"fmt"
	"strconv"
	"time"

	"github.com/erneap/metrics-api/models/interfaces"
)

type CreateMission struct {
	MissionDate    time.Time `json:"missionDate"`
	PlatformID     string    `json:"platformID"`
	SortieID       uint      `json:"sortieID"`
	Exploitation   string    `json:"exploitation"`
	PrimaryDCGS    string    `json:"primaryDCGS"`
	Communications string    `json:"communications"`
	TailNumber     string    `json:"tailNumber"`
	Overlap        uint      `json:"overlap"`
	Executed       bool      `json:"executed"`
	Aborted        bool      `json:"aborted"`
	Cancelled      bool      `json:"cancelled"`
	IndefDelay     bool      `json:"indefDelay"`
	Sensors        []string  `json:"sensors"`
}

type UpdateMission struct {
	ID             string `json:"id"`
	SensorID       string `json:"sensorID"`
	ImageTypeID    string `json:"imageTypeID"`
	ImageSubTypeID string `json:"imageSubTypeID"`
	Field          string `json:"field"`
	Value          string `json:"value"`
}

func (ur *UpdateMission) StringValue() string {
	return ur.Value
}

func (ur *UpdateMission) NumberValue() uint {
	num, err := strconv.ParseUint(ur.Value, 10, 32)
	if err != nil {
		fmt.Println(err.Error())
	}
	return uint(num)
}

func (ur *UpdateMission) BooleanValue() bool {
	answer, err := strconv.ParseBool(ur.Value)
	if err != nil {
		return false
	}
	return answer
}

func (ur *UpdateMission) DateValue() time.Time {
	answer, _ := time.ParseInLocation("2006-01-02", ur.Value, time.UTC)
	return answer
}

type MissionResponse struct {
	Missions  []interfaces.Mission `json:"missions"`
	Exception string               `json:"exception"`
}

File: client/src/app/models/metrics/mission.ts
import { IMissionData, IMissionSensor, MissionData, MissionSensor } from "./missionSensor";

export interface IMission {
    id?: string;
    missionDate: Date;
    platformID: string;
    sortieID: number;
    exploitation: string;
    tailNumber: string;
    communications: string;
    primaryDCGS: string;
    executed: boolean;
    cancelled: boolean;
    aborted: boolean;
    indefDelay: boolean;
    missionOverlap: number;
    comments: string;
    sensors: IMissionSensor[];
    _id?: string;
}

export class Mission implements IMission{
    public id?: string;
    public missionDate: Date;
    public platformID: string;
    public sortieID: number;
    public exploitation: string;
    public tailNumber: string;
    public communications: string;
    public primaryDCGS: string;
    public executed: boolean;
    public cancelled: boolean;
    public aborted: boolean;
    public indefDelay: boolean;
    public missionOverlap: number;
    public comments: string;
    public sensors: MissionSensor[];

    constructor(iMission?: IMission) {
        this.id = (iMission && iMission.id) 
            ? iMission.id : '';
        this.missionDate = (iMission && iMission.missionDate) 
            ? new Date(iMission.missionDate) : new Date(0);
        this.platformID = (iMission && iMission.platformID) 
            ? iMission.platformID : '';
        this.sortieID = (iMission && iMission.sortieID) 
            ? iMission.sortieID : 0;
        this.exploitation = (iMission && iMission.exploitation) 
            ? iMission.exploitation : 'Primary';
        this.tailNumber = (iMission && iMission.tailNumber) ? iMission.tailNumber : '';
        this.communications = (iMission && iMission.communications) 
            ? iMission.communications : 'LOS';
        this.primaryDCGS = (iMission && iMission.primaryDCGS) ? iMission.primaryDCGS : '';
        this.executed = (iMission && iMission.executed) ? true : false;
        this.cancelled = (iMission && iMission.cancelled) ? true : false;
        this.aborted = (iMission && iMission.aborted) ? true : false;
        this.indefDelay = (iMission && iMission.indefDelay) ? true : false;
        this.missionOverlap = (iMission && iMission.missionOverlap) 
            ? iMission.missionOverlap : 0;
        this.comments = (iMission && iMission.comments) ? iMission.comments : '';
        this.sensors = [];
        if (iMission && iMission.sensors && iMission.sensors.length > 0) {
            for (let i=0; i < iMission.sensors.length; i++) {
                let sen = iMission.sensors[i];
                this.sensors.push(new MissionSensor(sen));
            }
        }
    }
}